### PR TITLE
Add function to convert []interface -> []string

### PR DESCRIPTION
**Pull request by cesartw**
cuz we all hate `panic: interface conversion: interface {} is []interface {}, not []int64`

### CODE REVIEW

File: sliceutil/sliceutil.go [hunk 1]
```diff
@@ -214,3 +215,25 @@ func StringMap(list []string, f func(string) string) []string {
 
 	return ret
 }
+
+// InterfaceSliceTo converts []interface to any given slice.
+// It will ~optimistically~ try to convert interface item to the dst item type
+func InterfaceSliceTo(src []interface{}, dst interface{}) interface{} {
+	dstt := reflect.TypeOf(dst)
+	if dstt.Kind() != reflect.Slice {
+		panic("`t` is not an slice")
```
Comment (rafaeljusto) [inline sliceutil/sliceutil.go:224]:
```suggestion
		panic("`dst` is not an slice")
```

File: sliceutil/sliceutil.go [hunk 2]
```diff
@@ -214,3 +215,25 @@ func StringMap(list []string, f func(string) string) []string {
...
+	}
+
+	dstV := reflect.ValueOf(dst)
+	dstt.Elem()
```
Comment (rafaeljusto) [inline sliceutil/sliceutil.go:228]: Why do we need this? To panic if it's not an element?

**Review by ripexz — APPROVED** [review]

**Comment by rafaeljusto** [comment]
I think this would be easier to implement once we get Go 1.18 generics thing.

**Comment by cesartw** [comment]
> I think this would be easier to implement once we get Go 1.18 generics thing.

definitely

**Review by ready4god2513 — APPROVED** [review]

**Pull request merged**